### ISSUE TITLE
version 5.8 triggering quickFilter on delete 

### ISSUE THREAD

**Issue by FireScribeFire** [RealRaven2000/quickFilters#158]
Version 5.8 that updated today is triggering quickFilter's active filter assist every time I delete. I didn't see this problem before, but read version 5.7.1 was supposed to fix that, so whatever you changed screwed up your fix. I'm having to disable filter assist, which defeats the reason I have quick filters. Might as well delete the add-on if this continues.

**Comment by JeffreyELewis** [comment]
I 2nd this issue. I was using quickFilter on my laptop last week. I got a new desktop yesterday and set up Thunderbird. I downloaded the latest version of QuickFilter and am encountering the same issue. I did not have this issue on the laptop. Desktop is V5.8 and laptop is V5.7.1.

**Comment by molitar** [comment]
Yep this old bug has cropped back up.  I'm going to downgrade it's too buggy.

**Comment by RealRaven2000** [comment]
it's a feature, nit a bug.

change to the way quickFilters detects moves like deleting / archiving, see the change log.
you can disable delete in quickFilters settings, see here:

https://github.com/RealRaven2000/quickFilters/issues/145#issuecomment-1410970609

right click Filter assistant 
on the Actions page, click on Assistant should ignore the folders...
set

extensions.quickfilters.assistant.exclude.trash=true

this will disable the assistant when you delete Emails.

**Comment by RealRaven2000** [comment]
just to explain, 5.7.1 had a universal folder listener that reacted to all mail moves, as I was asked by Thunderbird staff to remove momkey patch code. then I had to do a lot of testing as it did trigger the assistant also by the filters themselves. I did disable filters for delete but also had to check whether an existing filter may have triggered the mail move. this lead to crashes or performance problems in a number of systems. 

I had to roll back to the old method (pre 5.6) but decided that it would be useful to be able to create filters when deleting / archiving mail ( not just via drag and drop) or putting mail in junk.

by default this is enabled for Delete and Junk but disabled  for Archiving.

**Comment by RealRaven2000** [comment]
> Yep this old bug has cropped back up. I'm going to downgrade it's too buggy.

Apart from this new feature, did you find any other bugs? 

Being able to create filters that delete mail is by design, and you can disable it with the new button on the filter actions page in quickFilters settings. Only affects users who have the filter assistant switched on permanently, but it's definitely more useful than only doing it when dragging mail to trash. See also #155 (Switching off quickFilters for delete and archive destroys my main use case) for a deeper discussion.

If you still want to revert to an older version I highly recommend 5.6 (which still uses the monkey patched code) as the code in 5.7.* can lead to performance problems or even crashes on some systems. That's because it needs to check all filters on every email move whether _they_ may have triggered the mail to be moved, which is a really bad idea in praxis.

![image](https://user-images.githubusercontent.com/6314674/219310676-03d346d5-a268-4d47-b648-7d403d48c5e5.png)

set `extensions.quickfilters.assistant.exclude.trash = true`

**Comment by McBobb** [comment]
Just a suggesting, but wouldn't it have made more sense to have the Junk and Trash folders excluded by default, like the Archives, and include instructions for adding them to the release notes instead?

**Comment by RealRaven2000** [comment]
yeah I thought about that but (case in point) people never read the release
notes.  since most people also do not have the assistant on all the time it
makes more sense like this to me.

 Normally you would switch on the assistant with the intent to create new
filters (or add rules to a filter that may also delete mail as a possible
action) - this way you can discover the feature without any learning curve.

I believe this default will benefit the larger number of users, rather than
the few people who run the assistant constantly - to switch  on auto
assistant you are on the same settings page (actions), which is something
an advanced user might enable.

I also added Screenshots to the version history page at quickfolders.org
this morning just to clarify:

https://quickfilters.quickfolders.org/version.html#5.18

timing is a little crunched a I am en route to the hospital.

**Comment by FireScribeFire** [comment]
I would've thought also that it would be off by default since I'm constantly moving mail and updating filters and I suspect a lot of others do the same, but at least there's a switch. Thanks for the explanation and instructions. Hope all ends well medically.

**Comment by JeffreyELewis** [comment]
> it's a feature, not a bug.
> 
> change to the way quickFilters detects moves like deleting / archiving, see the change log. you can disable delete in quickFilters settings, see here:
> 
> [#145 (comment)](https://github.com/RealRaven2000/quickFilters/issues/145#issuecomment-1410970609)
> 
> right click Filter assistant on the Actions page, click on Assistant should ignore the folders... set
> 
> extensions.quickfilters.assistant.exclude.trash=true
> 
> this will disable the assistant when you delete Emails.

RealRaven2000, your advise works great. Sorry I did not see this earlier.

Cheers

**Comment by RealRaven2000** [comment]
can you check if this is still the case on your system, will close the issue - feel free to reopen it
